src/api/searchingCity.ts

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios';
+import {
+    SearchCityTypes,
+    SearchCityCircle,
+    SearchCityCountryCode,
+    SearchCityLimit,
+    SearchCityLang,
+} from '../utils/constans';
+
+interface City {
+    id: string;
+    language: string;
+    localityType: string;
+    resultType: string;
+    title: string;
+    highlights: object;
+    address: {
+        city: string;
+        countryCode: string;
+        countryName: string;
+        county: string;
+        label: string;
+        state: string;
+    };
+}
+
+async function searchingCity(search: string) {
+    const path = `${process.env.REACT_APP_HERE_API_BASE_URL}autocomplete`;
+    try {
+        const {
+            data: { items },
+        } = await axios.get(path, {
+            params: {
+                q: search,
+                apiKey: process.env.REACT_APP_API_KEY,
+                types: SearchCityTypes.CITY,
+                in: `circle:${SearchCityCircle.CIRCLE};r=${SearchCityCircle.RADIUS}`,
+                in: `countryCode:${SearchCityCountryCode.COUNTRY_CODE}`,
+                limit: SearchCityLimit.LIMIT,
+                lang: SearchCityLang.LANG,
+            },
+        });
+        return items.map((item: City) => item.address.city);
+    } catch (error) {
+        alert(`Ошибка ${error.name} : ${error.message}`);
+        return error;
```

Comment:
may be would be better to return empty array? because you do not take into account in the app.tsx that an error will be returned.